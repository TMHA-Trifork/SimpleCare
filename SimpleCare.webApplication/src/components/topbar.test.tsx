import { describe, it, expect, vi } from "vitest";
import { renderWithRouter } from "../test/test-utils.tsx";
import TopBar from "./Topbar";

describe("TopBar", () => {
  it("should match snapshot", async () => {
    const { container } = renderWithRouter(<TopBar />);

    await vi.waitFor(() => {
      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  it("should render navigation links", async () => {
    const { container } = renderWithRouter(<TopBar />);

    await vi.waitFor(
      () => {
        const navigationLinks = container.querySelectorAll(
          "nav a, .nav-item a, a[href]"
        );
        expect(navigationLinks.length).toBeGreaterThan(0);
      },
      {
        timeout: 2000,
      }
    );
  });
});
