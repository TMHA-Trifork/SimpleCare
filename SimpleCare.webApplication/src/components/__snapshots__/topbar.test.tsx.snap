// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopBar > should match snapshot 1`] = `"<div class="topbar-container"><div class="topbar-logo">SimpleCare</div><div class="topbar-links"><a class="topbar-link" href="/">Home</a><a class="topbar-link" href="/emergency-ward">Emergency Ward</a><a class="topbar-link" href="/bed-ward">Bed Ward</a></div></div>"`;
